server/src/com/cloud/api/ApiResponseGsonHelper.java

```diff
@@ -27,30 +29,40 @@
 import com.google.gson.GsonBuilder;
 
 /**
- * The ApiResonseGsonHelper is different from ApiGsonHelper - it registeres one more adapter for String type required for api response encoding
+ * The ApiResonseGsonHelper is different from ApiGsonHelper - it registers one more adapter for String type required for api response encoding
```

Comment:
I have to read more on the gson changes you are referring to. But I think we shouldn't mix multiple things with this. The primary purpose of this PR is to fix the perf. issue in list API responses.

There may be some rework if gson is upgraded but it is better to do that when gson upgrade work is done.